sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient.cs

```diff
@@ -318,5 +289,35 @@ public virtual FormTrainingClient GetFormTrainingClient()
         }
 
         #endregion Training client
+
+        /// <summary>
+        /// Used as part of argument validation. Detects the <see cref="ContentType"/> of a stream and
+        /// throws an <see cref="ArgumentException"/> in case of failure.
+        /// </summary>
+        /// <param name="stream">The stream to which the content type detection attempt will be performed.</param>
+        /// <param name="paramName">The original parameter name of the <paramref name="stream"/>. Used to create exceptions in case of failure.</param>
+        /// <returns>The detected <see cref="ContentType"/>.</returns>
+        /// <exception cref="ArgumentException">Happens when detection fails or cannot be performed.</exception>
+        private static ContentType DetectContentType(Stream stream, string paramName)
+        {
+            ContentType contentType;
+
+            if (!stream.CanSeek)
+            {
+                throw new ArgumentException("Content type cannot be detected because stream is not seekable.", paramName);
+            }
+
+            if (!stream.CanRead)
+            {
+                throw new ArgumentException("Content type cannot be detected because stream is not readable.", paramName);
+            }
+
+            if (!stream.TryGetContentType(out contentType))
+            {
+                throw new ArgumentException("Content type of the stream could not be detected.", paramName);
```

Comment:
Was this the outcome of your discussion with our deputy architect?

Reply:
Note: after some internal discussion, we decided to keep it as an `ArgumentException` for the time being, but the possibility of throwing a `NotSupportedException` will be discussed as well.